apps/glusterfs/volume_entry_allocate.go

```diff
@@ -63,6 +64,165 @@ func (v *VolumeEntry) allocBricksInCluster(db *bolt.DB,
 	}
 }
 
+func (v *VolumeEntry) replaceBrickInVolume(db *bolt.DB, executor executors.Executor,
+	allocator Allocator,
+	oldBrickEntry *BrickEntry) (e error) {
+
+	logger.Info("Entered replace brick")
+	// Do the work in the database context so that the cluster
+	// data does not change while determining brick location
+	err := db.Update(func(tx *bolt.Tx) error {
+		brickids := v.BricksIds()
+		for _, brickid := range brickids {
+			brick, err := NewBrickEntryFromId(tx, brickid)
+			if err != nil {
+				return err
+			}
+
+			if brick.Info.NodeId != oldBrickEntry.Info.NodeId {
+				node, err := NewNodeEntryFromId(tx, brick.Info.NodeId)
+				if err != nil {
+					return err
+				}
+				logger.Info("Remove node %v from allocator", node.Info.Id)
+				err = node.removeAllDisksFromRing(tx, allocator)
+				if err != nil {
+					return err
+				}
+			}
+		}
+		defer func() {
```

Comment:
This may not work since the db.Update() is called in a db.Update().  It should definitely be done another way.